File: Amazecarenewvscode/src/main/java/com/example/demo/Service/DoctorService.java
package com.example.demo.Service;

import java.util.List;
import com.example.demo.dto.AppointmentDTO;
import com.example.demo.dto.DoctorDTO;
import com.example.demo.dto.MedicalRecordDTO;
import com.example.demo.dto.PrescriptionDTO;

public interface DoctorService {

    List<AppointmentDTO> viewDoctorAppointments(Long doctorId);

    DoctorDTO getDoctorByUsername(String username);

    List<AppointmentDTO> getUpcomingAppointments(Long doctorId);

    List<AppointmentDTO> getCompletedAppointments(Long doctorId);

    AppointmentDTO confirmAppointment(Long appointmentId);

    AppointmentDTO rejectAppointment(Long appointmentId);

    AppointmentDTO completeAppointment(Long appointmentId);

    MedicalRecordDTO addConsultationDetails(Long appointmentId, MedicalRecordDTO medicalRecord);

    PrescriptionDTO addPrescription(Long medicalRecordId, PrescriptionDTO prescription);

    List<MedicalRecordDTO> viewPatientHistory(Long patientId);
}

File: frontend/src/services/doctorService.js
import api from './api';

const doctorService = {
    getDoctorProfile: (username) => {
        return api.get(`/doctor/profile/${username}`).then(response => {
            return response.data;
        }).catch(error => {
             console.error("Failed to load secure doctor profile via backend API:", error);
             throw error;
        });
    },
    getAppointments: (doctorId) => {
        return api.get(`/doctor/${doctorId}/appointments`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    viewUpcomingAppointments: (doctorId) => {
        return api.get(`/doctor/${doctorId}/upcoming`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    viewCompletedAppointments: (doctorId) => {
        return api.get(`/doctor/${doctorId}/completed`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    confirmAppointment: (appointmentId) => {
        return api.put(`/doctor/appointment/${appointmentId}/confirm`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    rejectAppointment: (appointmentId) => {
        return api.put(`/doctor/appointment/${appointmentId}/reject`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    completeAppointment: (appointmentId) => {
        return api.put(`/doctor/appointment/${appointmentId}/complete`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    updateAppointmentStatus: (appointmentId, status) => {
        return api.put(`/appointments/${appointmentId}/status?status=${status}`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    addConsultationDetails: (appointmentId, recordDTO) => {
        return api.post(`/doctor/appointment/${appointmentId}/consultation`, recordDTO)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    addPrescription: (recordId, prescriptionDTO) => {
        return api.post(`/doctor/medicalrecord/${recordId}/prescription`, prescriptionDTO)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    viewPatientHistory: (patientId) => {
        return api.get(`/doctor/patient/${patientId}/history`)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    addMedicalRecord: (recordData) => {
        return api.post(`/medical-records`, recordData)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    updateMedicalRecord: (id, recordDTO) => {
        return api.put(`/medical-records/${id}`, recordDTO)
            .then(response => response.data)
            .catch(error => { throw error; });
    },
    getMedicalRecordByAppointment: (appointmentId) => {
        return api.get(`/medical-records/appointment/${appointmentId}`)
            .then(response => response.data)
            .catch(error => { throw error; });
    }
};

export default doctorService;
